File: challenges/leetcode/twosum/twosum.php
<?php

class Solution
{

    function twoSum($nums, $target)
    {
        $hm = [];
        for ($i = 0; $i < count($nums); $i++) {
            $c = $target - $nums[$i];

            if (array_key_exists($c, $hm)) {
                return [$hm[$c], $i];
            }
            $hm[$nums[$i]] = $i;
        }
        return [];
    }
}

File: leetcode/golang/twoSum.go
package main

func twoSum(nums []int, target int) []int {
	hm := make(map[int]int)
	for i := 0; i < len(nums); i++ {
		compl := target - nums[i]
		if id, ok := hm[compl]; ok {
			return []int{i, id}
		}

		hm[nums[i]] = i
	}
	return []int{}
}
